osquery/tables/system/hash.cpp

```diff
@@ -139,7 +158,108 @@ std::string hashFromFile(HashType hash_type, const std::string& path) {
   }
 }
 
-namespace tables {
+/*
+ * FileHashCache implements persistent in-memory caching for files' hashes.
+ * this cache has LRU eviction policy. the hash is recalculated
+ * every time the mtime or size of the file changes.
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#if defined(WIN32)
+
+typedef struct _stat Stat;
+#define xstat(path, st) _stat((path), static_cast<struct _stat *>(st))
+
+#else
+
+#include <unistd.h>
+
+typedef struct stat Stat;
+#define xstat(path, st) stat((path), static_cast<struct stat *>(st))
+
+#endif
+
+struct FileHashCache {
+  time_t mtime; // last modification time of the file, # sec since UNIX
```

Comment:
For these local variable names, perhaps you could change their naming to something more descriptive like `last_mtime` and avoid needing the comment altogether? For most of the project it's a pretty safe assumption that a timestamp is the UNIX epoch timestamp.

If you'd still like that comments feel free to give them their own line.

Reply:
addressed.